lightbluetent/users.py

```diff
@@ -123,8 +123,12 @@ def register():
     crsid = auth_decorator.principal
 
     existing_user = User.query.filter_by(crsid=crsid).first()
-    if existing_user:
-        return redirect(url_for("users.home"))
+
+    if existing_user is not None:
+
+        # Check that the user's not a dummy user without a name and email address.
+        if existing_user.email is not None and existing_user.full_name is not None:
```

Comment:
Sorry, see my above edits.

Reply:
Ah, I see. I'm not sure how I feel about creating a full blown user for that purpose. I also think it is useful to separate users that create rooms vs "users" that go to rooms. Perhaps a new Visitor table with just an id, crsid and the whitelist to which it belongs?

Reply:
Say if you want to query all users of the platform itself rather than visitors. That is useful to know. Eg. we have 400 users of Greenlight but I know all of those have explicitly signed up.